server/src/api/modules/groups/services/EditGroupService.ts

```diff
@@ -448,3 +479,19 @@ function calculateRoleChangeMaps(
 }
 
 export { editGroup };
+
+async function updateGroupRoleOrder(
+  groupId: number,
+  roleOrders: Array<{ role: GroupRole; index: number }>,
+  transaction: PrismaTypes.TransactionClient
+) {
+  const orderWithGroupId = roleOrders.map(x => ({ ...x, groupId: groupId }));
+
+  await transaction.groupRoleOrder.deleteMany({
+    where: { groupId }
+  });
+
+  await transaction.groupRoleOrder.createMany({
+    data: orderWithGroupId
```

Comment:
I'd just do `data: roleOrders.map(x => ({ ...x, groupId: groupId }))` and skip creating that variable altogether